### PR TITLE
Extract prot schema DDL into prot.sql scripts

### PR DESCRIPTION
**Pull request by labkey-adam**
#### Rationale
We're planning to split protein schema management into its own module; this is a step in that direction. [Secure Issue 49958: Split protein schema and code from ms2 module](https://www.labkey.org/home/Developer/issues/Secure/issues-details.view?issueId=49958)

### CODE REVIEW

File: ms2/resources/schemas/dbscripts/postgresql/ms2-0.000-21.000.sql [hunk 1]
```diff
@@ -416,6 +87,8 @@ ALTER TABLE ms2.Fractions ADD COLUMN MS2ScanCount INT;
 ALTER TABLE ms2.Fractions ADD COLUMN MS3ScanCount INT;
 ALTER TABLE ms2.Fractions ADD COLUMN MS4ScanCount INT;
 
+UPDATE ms2.fractions SET MzXmlUrl = 'file:///' || substr(MzXmlUrl, 7) WHERE MzXmlUrl LIKE 'file:/_%' AND MzXmlUrl NOT LIKE 'file:///%' AND MzXmlUrl IS NOT NULL;
```
Comment (labkey-jeckels) [inline ms2/resources/schemas/dbscripts/postgresql/ms2-0.000-21.000.sql]: Feel free to delete this line - the table will always be empty when bootstrapping.
```suggestion
```

File: ms2/resources/schemas/dbscripts/sqlserver/ms2-0.000-21.000.sql [hunk 2]
```diff
@@ -391,6 +81,14 @@ CREATE TABLE ms2.Fractions
 CREATE INDEX IX_Fractions_Run_Fraction ON ms2.Fractions(Run, Fraction);
 CREATE INDEX IX_Fractions_MzXMLURL ON ms2.fractions(mzxmlurl);
 
+ALTER TABLE ms2.Fractions ADD ScanCount INT
+ALTER TABLE ms2.Fractions ADD MS1ScanCount INT
+ALTER TABLE ms2.Fractions ADD MS2ScanCount INT
+ALTER TABLE ms2.Fractions ADD MS3ScanCount INT
+ALTER TABLE ms2.Fractions ADD MS4ScanCount INT
+
+UPDATE ms2.fractions SET MzXmlUrl = 'file:///' + substring(MzXmlUrl, 7, 400) WHERE MzXmlUrl LIKE 'file:/_%' AND MzXmlUrl NOT LIKE 'file:///%' AND MzXmlUrl IS NOT NULL;
```
Comment (labkey-jeckels) [inline ms2/resources/schemas/dbscripts/sqlserver/ms2-0.000-21.000.sql]:
```suggestion
```

File: ms2/src/org/labkey/ms2/MS2Module.java [hunk 3]
```diff
@@ -318,9 +318,9 @@ public Collection<String> getSummary(Container c)
 
     @Override
     @NotNull
-    public Set<String> getSchemaNames()
+    public List<String> getSchemaNames()
     {
-        return PageFlowUtil.set(MS2Schema.SCHEMA_NAME, ProteinManager.getSchemaName());
+        return List.of(ProteinManager.getSchemaName(), MS2Schema.SCHEMA_NAME);
```
Comment (labkey-jeckels) [inline ms2/src/org/labkey/ms2/MS2Module.java:323]: I assume this ensures that `prot` is created first so it can be referenced by FKs from `ms2`?

**Review by labkey-jeckels — APPROVED** [review]

File: ms2/src/org/labkey/ms2/MS2Module.java (hunk 3 above)
Comment (labkey-adam) [reply ms2/src/org/labkey/ms2/MS2Module.java:323]: Yep, `DefaultModule.versionUpdate()` enumerates and upgrades schemas in the order provided by `FileSqlScriptProvider.getSchemas()`, which enumerates `getSchemaNames()`.

**Pull request merged**